pkg/virt-handler/device-manager/device_controller.go

```diff
@@ -233,38 +272,56 @@ func (c *DeviceController) refreshPermittedDevices() {
 	// This function can be called multiple times in parallel, either because of multiple
 	//   informer callbacks for the same event, or because the configmap was quickly updated
 	//   multiple times in a row. To avoid starting/stopping device plugins multiple times,
-	//   we need to protect c.devicePlugins, which we read from in
+	//   we need to protect c.startedPlugins, which we read from in
 	//   c.updatePermittedHostDevicePlugins() and write to below.
-	c.devicePluginsMutex.Lock()
+	c.startedPluginsMutex.Lock()
+	defer c.startedPluginsMutex.Unlock()
 
 	enabledDevicePlugins, disabledDevicePlugins := c.updatePermittedHostDevicePlugins()
 
 	// start device plugin for newly permitted devices
 	for resourceName, dev := range enabledDevicePlugins {
-		go c.startDevicePlugin(dev)
-		c.devicePlugins[resourceName] = dev
+		c.addAndStartDevice(resourceName, dev)
```

Comment:
this method clearly does 2 things :) 

I'd just name it `startDevice` - I consider the side effect of adding it to the cache an implementation detail.